### PR TITLE
Update app termination by Ctrl-C.

### PR DESCRIPTION
**Pull request by nk0leg**
Solution based on a ["Signals and Signal Handling"](https://msdn.microsoft.com/en-us/library/ms811896.aspx#ucmgch09_topic3)

### CODE REVIEW

**Comment by nk0leg** [comment]
@pestOO, @malirod, @anosach-luxoft  please review

**Comment by nk0leg** [comment]
@pestOO, @malirod, @anosach-luxoft please review

**Comment by nk0leg** [comment]
@pestOO, @malirod , @anosach-luxoft  pull request update
please see 589f8f7

**Comment by anosach-luxoft** [comment]
Reviewed

File: src/components/utils/src/signals_linux.cc [hunk 1]
```diff
@@ -37,8 +37,29 @@
...
+    default:
+      ::utils::handleSigs(NULL, "Unexpected signal has been caught");
+      break;
+  }
+}
+}  //  namespace
+
 namespace utils {
 
+void handleSigs(void* signal_handle, const char* log_name) {}
```
Comment (anosach-luxoft) [inline src/components/utils/src/signals_linux.cc]: @nk0leg wrong function name

File: src/components/utils/src/signals_linux.cc [hunk 2]
```diff
@@ -37,8 +37,29 @@
...
+void WaitForSdlExecute() {}
+
 bool SubscribeToInterruptSignal(sighandler_t func) {
```
Comment (anosach-luxoft) [inline src/components/utils/src/signals_linux.cc:42]: you've removed this functions from header so remove them from here or move to anonymous namespace

File: src/components/utils/src/signals_linux.cc [hunk 3]
```diff
@@ -37,8 +37,29 @@
...
+      ::utils::handleSigs(NULL, "Unexpected signal has been caught");
+      break;
+  }
+}
+}  //  namespace
+
 namespace utils {
 
+void handleSigs(void* signal_handle, const char* log_name) {}
+void WaitForSdlExecute() {}
```
Comment (anosach-luxoft) [inline src/components/utils/src/signals_linux.cc]: I think you have to call `pause()` here

File: src/components/utils/src/signals_linux.cc (hunk 1 above)
Comment (nk0leg) [inline src/components/utils/src/signals_linux.cc]: see 5b6e760

File: src/components/utils/src/signals_linux.cc (hunk 2 above)
Comment (nk0leg) [inline src/components/utils/src/signals_linux.cc:42]: see 5b6e760

File: src/components/utils/src/signals_linux.cc (hunk 3 above)
Comment (nk0leg) [inline src/components/utils/src/signals_linux.cc]: see 5b6e760

**Pull request merged**